### DISCUSSION TITLE
Use streams during CSV import

### DISCUSSION THREAD

**Discussion by vincius0000** [enrique-lozano/Monekin#233] [Ideas / New features]
Now during CSV import a whole CSV document is loaded in memory. It should be changed to take in a batch or single line and that can be achieved by using streams.

This is to keep strain away from mobile devices on large CSV files (if there were to be any).

**Comment by enrique-lozano** [comment]
At the moment the app is just starting and I don't think many users use this feature. Still, I think what you say makes a lot of sense and we should keep an eye on it.

The truth is that I haven't tested the performance with a very large CSV, I have no idea how bad it will be. I will do tests in the future if I don't have more information on this aspect and the app continues to grow.

**Reply by gursuj** [reply]
I think the app took around 10-15 seconds to import a CSV with 570 rows (33kb file). Not really a complaint, just wanted to share. Also, my phone is somewhat of a budget phone.

Edit: Had to delete all data and re-import the same CSV. Didn't time it but I think it took almost a minute

**Reply by enrique-lozano** [reply]
Wow, that seems a lot for me. I think it's time to take a look into that 🕵️‍♂️
